core/src/main/scala/chisel3/properties/Property.scala

```diff
@@ -298,29 +314,58 @@ sealed trait Property[T] extends Data { self =>
 
 }
 
+trait ClassTypeProvider[A] {
```

Comment:
```suggestion
sealed trait ClassTypeProvider[A] {
```
This should be sealed right?

Reply:
make this `sealed` and `private[chisel3]`